src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -100,12 +85,33 @@ private ArgumentCaptor(Class<? extends T> clazz) {
      * @return null or default values
      */
     public T capture() {
-        Mockito.argThat(capturingMatcher);
+        registerCapturingMatcher(capturingMatcher);
         return handyReturnValues.returnFor(clazz);
     }
 
     /**
+     * Use it to capture the argument that satisfies the matcher. This method <b>must be used inside of verification</b>.
+     * <p>
+     * Internally, this method registers a special implementation of an {@link ArgumentMatcher}.
+     * This argument matcher stores the argument value that satisfies the matcher so that you can use it later to perform assertions.  
+     * <p>
+     * See examples in javadoc for {@link ArgumentCaptor} class.
+     * 
+     * @return null or default values
+     */
+    public T captureIf(ArgumentMatcher matcher) {
+        capturingMatcher = CapturingMatcher.is(matcher, arguments);
```

Comment:
I think the assignment is not needed cause the implementation can access the `arguments`-List to get the captured values.